src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+﻿using System;
+using System.Drawing;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Globalization;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// This is a high-level interface to an LCD display. 
+    /// It supports automatic wrapping of text, automatic scrolling and code page mappings.
+    /// This class is thread safe, however using Write from different threads may lead to unexpected results, since the order is not guaranteed. 
+    /// </summary>
+    public sealed class LcdConsole : IDisposable
+    {
+        private ICharacterLcd _lcd;
+
+        /// <summary>
+        /// The text currently on the display (required for arbitrary scrolling)
+        /// </summary>
+        private StringBuilder[] _currentData;
+        private LineFeedMode _lineFeedMode;
+        private readonly object _lock;
+        private readonly bool _shouldDispose;
+        private TimeSpan _newLineSleepTime;
+        private string _romType;
+        private Encoding _characterEncoding;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="LcdConsole"/> class using the specified LCD low-level interface. 
+        /// This class automatically configures the low-level interface. Do not use the low-level interface at the same time. 
+        /// </summary>
+        /// <param name="lcd">The low-level LCD interface.</param>
+        /// <param name="romType">Name of character ROM of display. Currently supported types: A00 and A02.</param>
+        /// <param name="shouldDispose">If the class should dispose the LCD driver when it is disposed. Defaults to true</param>
+        public LcdConsole(ICharacterLcd lcd, String romType, bool shouldDispose = true)
```

Comment:
`string`. Can you clarify what `ROM` means? Read-Only Memory? If so I cannot correlate how it fits in this API

Reply:
@krwq Yes, it means "Read only memory". Its the fixed part of the character memory of the display. For the HD44780, there are two documented ROM types, "A00", which contains japanese characters in the upper 127 bits of the character table, and "A02" which contains european dyacritics (äàÄ etc) there. I used a string instead of an enum because other LCD displays might contain other character maps not known here and this class should work against any implementation of ICharcterLcd.